### ISSUE TITLE
quarkus and classpath hell

### ISSUE THREAD

**Issue by nimo23** [quarkusio/quarkus#5870]
**Describe the bug**
Compiling the app shows the following warning:

```
[WARNING] [io.quarkus.creator.curator.Curator] These dependencies are not recommended:
	- jakarta.ws.rs:jakarta.ws.rs-api should be replaced by org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec
	- jakarta.xml.bind:jakarta.xml.bind-api should be replaced by org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec
	- javax.activation:activation should be replaced by com.sun.activation:jakarta.activation
	- javax.activation:javax.activation-api should be replaced by jakarta.activation:jakarta.activation-api
	- javax.enterprise:cdi-api should be replaced by jakarta.enterprise:jakarta.enterprise.cdi-api
	- javax.inject:javax.inject should be replaced by jakarta.inject:jakarta.inject-api
	- javax.validation:validation-api should be replaced by jakarta.validation:jakarta.validation-api
	- javax.ws.rs:javax.ws.rs-api should be replaced by org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec
	- javax.xml.bind:jaxb-api should be replaced by org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec
You might end up with two different versions of the same classes or with an artifact you shouldn't have in your classpath.
```
Is this a kind of classpath-hell? I did not include these libs by myself..maybe some other 3rd party libs inlcude it? Must I really look for all my 3rd party dependencies which one integrates which conflicting lib to exclude them manually in my pom.xml? 

**Expected behavior**
quarkust should not include them in the lib while compiling, if it already has them in the classpath.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 1.0 Final

**Comment by dmlloyd** [comment]
We should automatically map these dependencies, instead of warning about them, IMO.

**Comment by mrasante** [comment]
@dmlloyd any news about when this fix might be implemented? This is turning out to be a show-stopper for a potential quarkus adoption.

**Comment by gsmet** [comment]
@mrasante It should be fixed in 1.3 alphas. But this issue is very vague so if you still have an issue with 1.3.0.Alpha2, please open a new issue with all the details.

**Comment by stuartwdouglas** [comment]
This is fixed now.
